src/providers/mod.rs

```diff
@@ -192,7 +192,7 @@ async fn setup_machine(
     let mut m = m.connect_ssh(username, private_key, max_wait, 22).await?;
 
     tracing::debug!("setting up instance");
-    f(&mut m.ssh).await.wrap_err("setup procedure failed")?;
+    f(&mut m).await.wrap_err("setup procedure failed")?;
```

Comment:
The setup function won't work if there is no ssh, so the existence of the setup functions should probably go away if the openssh feature is disabled.

Reply:
I believe it does work. A setup function that uses `Machine.ssh` won't compile unless the `openssh` feature is enabled.

Reply:
You are right, I meant to say that it won't be useful, but as you say it is possible to use a different connection method.